vignettes/develop.Rmd

```diff
@@ -0,0 +1,327 @@
+---
+title: "Developing odbc"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Developing odbc}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette is intended to help developers of the R package install needed dependencies.
+
+While the odbc package contains some documentation on how to install and configure database drivers in `vignette("setup")`, the documentation assumes that users are connecting to databases that have already been set up. In order to test package functionality, though, odbc sets up small example database deployments.
+
+For the most part, this vignette assumes a MacOS system with aarch64 (e.g. M1 or M2) architecture. For Linux example code, see [`.github/workflows/db.yaml`](https://github.com/r-dbi/odbc/blob/main/.github/workflows/db.yaml), and for Windows, see [`.github/workflows/db-windows.yml`](https://github.com/r-dbi/odbc/blob/main/.github/workflows/db-windows.yml).
+
+## PostgreSQL
+
+On MacOS, install PostgreSQL with:
+
+```shell
+brew install postgresql@14
+
+```
+
+You'll also need to install and configure the PostgreSQL driver `psqlodbc`; see `vignette("setup")` to learn more.
+
+To launch a PostgreSQL server locally, run:
+
+```
+brew services start postgresql@14
+```
+
+Next, create a database called "test" (or by whatever name is in the entry `Database` in your `odbc.ini` file):
+
+```shell
+createdb test
+```
+
+At this point, you should be able to connect to PostgreSQL through the R interface. Connect with:
+
+```r
+postgres <- dbConnect(odbc(), "PostgreSQL")
+```
+
+where `"PostgreSQL"` is replaced with whatever DSN you've configured.
+
+
+## MySQL
+
+First, installing MySQL with Homebrew:
+
+```shell
+brew install mysql@8.2
+```
+
+Available installers of the MySQL drivers (called the Connector/ODBC for MySQL) on Unix systems assume the iODBC driver manager. The driver thus needs to be installed from source.
```

Comment:
I discovered a slightly hacky way to make this much easier — `brew install mariadb-connector-odbc`. MariaDB is a fork of MySQL (after Oracle bought) and the ODBC driver appears to be compatible.

Reply:
Sooo much easier! [c459dd9](https://github.com/r-dbi/odbc/pull/670/commits/c459dd9cc6b5fe765d44f73e38965405682f5118).